File: hackerrank/30 days of code/day 11, 2d arrays/day_11_2d_arrays.py
#!/bin/python3

if __name__ == "__main__":

    arr = []

    for _ in range(6):
        arr.append(list(map(int, input().rstrip().split())))

    maxSum = -9 * 7
    for i in range(1, 5):
        for j in range(1, 5):

            currentSum = (
                arr[i - 1][j - 1]
                + arr[i - 1][j]
                + arr[i - 1][j + 1]
                + arr[i][j]
                + arr[i + 1][j - 1]
                + arr[i + 1][j]
                + arr[i + 1][j + 1]
            )

            if currentSum > maxSum:
                maxSum = currentSum

    print(maxSum)

File: hackerrank/30 days of code/day 11, 2d arrays/day_11_2d_arrays.rb
#!/bin/ruby

arr = Array.new(6)

6.times do |i|
    arr[i] = gets.rstrip.split.map(&:to_i)
end

maxSum = -9 * 7
for i in 1..4
    for j in 1..4

        currentSum = arr[i-1][j-1] + arr[i-1][j] + arr[i-1][j+1] + arr[i][j] + arr[i+1][j-1] + arr[i+1][j] + arr[i+1][j+1]

        if currentSum > maxSum
            maxSum = currentSum
        end
    end
end
puts maxSum